### PR TITLE
Add an API flag to restore_secrets script

### PR DESCRIPTION
**Pull request by DellaBitta**
Whenever I do local development I run restore_secrets, which updates all of the Google Services data for all of the products. This pollutes my local branch since normally I'm only working on one product.

This change adds an optional `api` flag so that you can checkout the secrets for only one product at a time.

***

- Local testing.
- [Integration Test CI](https://github.com/firebase/firebase-cpp-sdk/actions/runs/2502019096)

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [X] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: scripts/gha/restore_secrets.py [hunk 1]
```diff
@@ -51,6 +53,7 @@
 flags.DEFINE_string("passphrase_file", None,
                     "Path to file with passphrase. Use \"-\" (without quotes) for stdin.")
 flags.DEFINE_string("artifact", None, "Artifact Path, google-services.json will be placed here.")
+flags.DEFINE_string("api", None, "Retrieve secret for the particular SDK only.")
```
Comment (sunmou99) [inline scripts/gha/restore_secrets.py:56]: Good idea. And what about make it a list of api?
`flags.DEFINE_list("apis", None, "Retrieve secret for the particular SDKs only.")`

Comment (DellaBitta) [reply scripts/gha/restore_secrets.py:56]: Is there a usecase for this?  Generally we're only working on one API/SDK at a time, or all at the same time in the case of CI.

Comment (sunmou99) [reply scripts/gha/restore_secrets.py:56]: e.g. When someone tries to build only the failed testapps locally.

**Review by sunmou99 — APPROVED** [review]

File: scripts/gha/restore_secrets.py (hunk 1 above)
Comment (DellaBitta) [reply scripts/gha/restore_secrets.py]: Done!  PTAL!

**Pull request merged**